app/javascript/contentDisplayPolicy/initHiddenComments.js

```diff
@@ -42,20 +16,65 @@ For further actions, you may consider blocking this person and/or reporting abus
       });
   }
 
+  function showHideCommentsModal(commentId) {
+    const form = document.getElementById('hide-comments-modal__form');
+    form.action = `/comments/${commentId}/hide`;
+
+    window.Forem.showModal({
+      title: 'Confirm hiding the comment',
+      contentSelector: '#hide-comments-modal',
+      overlay: true,
+    }).then(() => {
+      const hideCommentForms = document.querySelectorAll(
+        '.hide-comments-modal__form',
+      );
+
+      hideCommentForms.forEach((form) => {
+        form.addEventListener('submit', handleHideCommentsFormSubmit);
+      });
```

Comment:
(suggestion / not-blocking) Am I right in thinking we need to do this because the form HTML is "copied" into the modal? If so I think you could likely just target the form that's inside the modal, e.g.

```suggestion
      const hideCommentForm = document.querySelector(
        '#window-modal .hide-comments-modal__form',
      );

      hideCommentForm.addEventListener('submit', handleHideCommentsFormSubmit);
```